addons/stock/models/stock_inventory.py

```diff
@@ -179,6 +192,9 @@ def _get_inventory_lines_values(self):
         locations = self.env['stock.location'].search([('id', 'child_of', [self.location_id.id])])
         domain = ' location_id in %s'
         args = (tuple(locations.ids),)
+        vals = []
+        Product = products = self.env['product.product']
```

Comment:
Why multiple assignment ? Products and products aren't different variables with equal values; they're different names for the same identical Python variables.

You can test it by trying 'Product is product' it will return true.